src/algorithms/time_evolution/evoltools.jl

```diff
@@ -1,28 +1,28 @@
 """
-    get_gate(dt::Float64, H::LocalOperator)
+    get_expham(dt::Float64, H::LocalOperator)
 
-Compute `exp(-dt * H)` from the nearest neighbor Hamiltonian `H`.
+Compute `exp(-dt * H)` from Hamiltonian `H`.
+Each term in `H` must be a single `TensorMap`.
 """
-function get_gate(dt::Float64, H::LocalOperator)
-    @assert all([
-        numin(op) == 2 && norm(Tuple(terms[2] - terms[1])) == 1.0 for (terms, op) in H.terms
-    ]) "Only nearest-neighbour terms allowed"
-    return LocalOperator(
-        physicalspace(H), Tuple(sites => exp(-dt * op) for (sites, op) in H.terms)...
-    )
+function get_expham(dt::Float64, H::LocalOperator)
+    return LocalOperator(H.lattice, (sites => exp(-dt * op) for (sites, op) in H.terms)...)
 end
 
 """
-    is_equivalent(
-        bond1::NTuple{2,CartesianIndex{2}},
-        bond2::NTuple{2,CartesianIndex{2}},
-        (Nrow, Ncol)::NTuple{2,Int},
-    )
+Check if a Hamiltonian contains only nearest neighbor terms
+"""
+function is_nnham(H::LocalOperator)
+    return all([
+        numin(op) == 2 && norm(Tuple(terms[2] - terms[1])) == 1.0 for (terms, op) in H.terms
+    ])
```

Comment:
```suggestion
    return all(H.terms) do (terms, op)
        return numin(op) == 2 && sum(abs, Tuple(terms[2] - terms[1])) == 1
    end
```
The anonymous function doesn't have to allocate this way, and I think it is safer to use `sum(abs, ...)` to avoid having to deal with floating point equalities.